test/controllers/trips_controller_test.rb

```diff
@@ -1,23 +1,106 @@
 require "test_helper"
 
 describe TripsController do
+  let (:new_trip) {
+    Trip.new(date: Time.now + 5.days, cost: 5)
+  }
+  let (:trip_hash) {
+    {
+      trip: {
+        date: Time.now + 6.days,
+        cost: 5,
+        rating: 4
+      }
+    }
+  }
   describe "show" do
-    # Your tests go here
+    it "can get a valid trip" do
+      trip = new_trip.save
+
+      # Act
+      get trip_path(trip.id)
+
+      # Assert
+      must_respond_with :success
+    end
+
+    it "will redirect for an invalid trip" do
+      # Act
+      get trip_path(-1)
+
+      # Assert
+      must_respond_with :redirect
+    end
   end
 
   describe "create" do
-    # Your tests go here
+    it "can create a new trip" do
```

Comment:
This test fails. Consider the trip `create` method. To create a trip you need the `:passenger_id` from the params and a `driver`. You make use of the nested route that can be viewed by typing `rails routes` 👍 
 `passenger_trips POST   /passengers/:passenger_id/trips(.:format)                                                trips#create` 

Consider how to write the test given this information and let me know if you have questions.

```
TripsController::create
  test_0001_can create a new trip                                 FAIL (0.01s)
        1.
        "Trip.count" didn't change
        test/controllers/trips_controller_test.rb:40:in `block (3 levels) in <main>'
```